libs/File2DProcessingTools/source/CSvgFileWriter.cpp

```diff
@@ -77,5 +62,93 @@ namespace File2DProcessingTools {
         return (double) _width / width < (double) _height / height ?
                (double) _width / width : (double) _height / height;
     }
+
+    void CSvgFileWriter::writeBeginning(std::ofstream &svg_file) {
+
+        svg_file << "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"yes\"?>\n";
+        svg_file << "<svg height = \"" << _height << "px\"  width = \""
+                 << _width << "px\">\n";
+
+    }
+
+    void CSvgFileWriter::writePoints(const File2DProcessingTools::CVectorGraphicsData &data,
+                                     std::ofstream &svg_file, std::pair<double, double> max,
+                                     std::pair<double, double> min) {
+        for (const auto &line : data.getLineSegments()) {
+            svg_file << "<path d=\"M "
+                     << scale * (line.getFirstPoint().getX() - min.first) << " "
+                     << scale * (max.second - min.second - (line.getFirstPoint().getY() - min.second))
+                     << " L "
+                     << scale * (line.getSecondPoint().getX() - min.first) << " "
+                     << scale * (max.second - min.second - (line.getSecondPoint().getY() - min.second))
+                     << " L\" stroke=\"black\" stroke-width=\"1\"  />\n";
+        }
+    }
+
+    void CSvgFileWriter::writeEnding(std::ofstream &svg_file) {
+        svg_file << "</svg>\n";
+    }
+
+    void CSvgFileWriter::checkFile(const std::string &filepath, std::ofstream &svg_file, bool write_force) {
+
+        if (!isFileExists(filepath, write_force))
+            throw File2DProcessingTools::Exceptions::EFileAlreadyExistsException(
+                    "Pass write_force = true to overwrite the existing file.");
+
+
+        svg_file.open(filepath);
+        if (!svg_file.is_open()) {
+            throw File2DProcessingTools::Exceptions::EFileCannotBeOverwritten("Couldn't be opened");
+        }
+
+        if (svg_file.bad()) {
```

Comment:
Checking badbit without writing attemp
You can do test writing or check is writing succeeded after writing all data, using [`.good()` ](https://en.cppreference.com/w/cpp/io/basic_ios/good)